addons/mass_mailing/controllers/main.py

```diff
@@ -55,3 +87,51 @@ def full_url_redirect(self, code, stat_id, **post):
 
         request.env['link.tracker.click'].add_click(code, request.httprequest.remote_addr, country_code, stat_id=stat_id)
         return werkzeug.utils.redirect(request.env['link.tracker'].get_url_from_code(code), 301)
+
+    @http.route('/mail/mailing/blacklist/check', type='json', auth='none')
+    def check_blacklist(self, email):
```

Comment:
Naming: I would name controller methods using black prefix: blacklist_check, blacklist_add, ... namespace will be great again that way.